src/proxy.sol

```diff
@@ -16,37 +16,65 @@
 
 pragma solidity >=0.6.0 <0.8.0;
 
-import { Title, TitleOwned, TitleLike } from "tinlake-title/title.sol";
+import "tinlake-auth/auth.sol";
 
 interface RegistryLike {
     function cacheRead(bytes memory _code) external view returns (address);
     function cacheWrite(bytes memory _code) external returns (address target);
 }
-// Proxy is a proxy contract that is controlled by a Title NFT (see tinlake-title)
-// The proxy execute methods are copied from ds-proxy/src/proxy.sol:DSProxy
-// (see https://github.com/dapphub/ds-proxy)
-contract Proxy is TitleOwned {
-    uint public         accessToken;
+
+contract Proxy is Auth {
+
+    mapping (address => uint256) public users;
+    address public target; // safe actions that can be called by users
+
     RegistryLike public registry;
 
-    constructor() TitleOwned(msg.sender) public {
-        registry = RegistryLike(msg.sender);
+    event Safe(address user);
+    event UserAdded(address user);
+    event UserRemoved(address user);
+
+    modifier user {
+        require(users[msg.sender] == 1, "tinlake/not-authorized");
+        _;
     }
 
-    function init(uint accessToken_) public {
-        // parameter accessToken_: always > 0
-        require(accessToken == 0);
-        accessToken = accessToken_;
+    constructor() {
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
     }
 
-    function execute(address _target, bytes memory _data)
+    function safe(address _target) external auth {
```

Comment:
I'd just change this to a `file(bytes32 what, address target)` method like https://github.com/centrifuge/connectors/blob/main/src/Connector.sol#L70, and just implement it for `target` as `what`